rhel_osv/osv.py

```diff
@@ -0,0 +1,174 @@
+    """
+    Class to hold event information for a Range. Advisories for Red Hat RPM based products always
+    assume all previous versions are affected.
+    """
+    INTRODUCED = "introduced"
+    FIXED = "fixed"
+
+    def __init__(self, event_type: str, version: str = "0"):
+        expected = (self.INTRODUCED, self.FIXED)
+        if event_type not in expected:
+            raise ValueError(f"Expected one of {expected} for type. Got {event_type}")
+        self.event_type = event_type
+        self.version = version
+
+    def encode_json(self):
+        """
+        Custom JSON encoding for event type which changes attribute name depending on the type of
+        event eg. introduced or fixed
+        """
+        if self.event_type == Event.INTRODUCED:
+            return {Event.INTRODUCED: self.version}
+        elif self.event_type == Event.FIXED:
+            return {Event.FIXED: self.version}
+        else:
+            raise ValueError("Unexpected event_type for Event")
+
+
+
+class Range(object):
+    """
+    Class to hold range information for a Package. Ecosystem here refers to RPM versions as defined
+    in https://github.com/rpm-software-management/rpm/blob/master/rpmio/rpmvercmp.c
+    """
+    def __init__(self, fixed: str):
+        self.type = "ECOSYSTEM"
+        self.events = [Event("introduced"), Event("fixed", fixed)]
+
+
+class Package(object):
+    """
+    Class to hold package data for an Affect. Expects an ecosystem string that starts with
+    CPE_PATTERN.
+    Replaces the CPE prefix 'redhat' part with 'Red Hat' to match more closely with other ecosystem
+    identifiers in the OSV database
+    """
+    CPE_PATTERN = re.compile(r"cpe:/[oa]:(redhat)")
+
+    def __init__(self, name: str, ecosystem: str, purl: str):
+        self.name = name
+        if not self.CPE_PATTERN.match(ecosystem):
+            raise ValueError(f"Got unsupported ecosystem: {ecosystem}")
+        self.ecosystem = f"Red Hat{self.CPE_PATTERN.split(ecosystem, maxsplit=1)[-1]}"
+        self.purl = purl
+
+
+class Affected(object):
+    """
+    Class to hold affected data for a Vulnerability
+    """
+    def __init__(self, remediation: csaf.Remediation):
+        self.package = Package(
+            remediation.component,
+            remediation.cpe,
+            remediation.purl
+        )
+        self.ranges = [Range(remediation.fixed_version)]
+
+
+class OSV(object):
+    """
+    Class to convert CSAF data to OSV
+    """
+    SCHEMA_VERSION = "1.6.3"
+    # Go package advisory reference prefix
+    PKG_GO_DEV_VULN = "https://pkg.go.dev/vuln/"
+    # Other common advisory prefixes in Red Hat Advisories
+    ADVISORY_URL_PREFIXES = (
+        PKG_GO_DEV_VULN,
+        "https://www.cve.org/CVERecord",
+        "https://nvd.nist.gov/vuln/detail/",
+        "https://www.kb.cert.org/vuls/id/",
+    )
+
+    def __init__(self, csaf: csaf.CSAF):
+        self.schema_version = self.SCHEMA_VERSION
+        self.id = csaf.id
```

Comment:
I note that in the converted output diff, the ID prefix is lowercase...

```suggestion
        self.id = csaf.id.upper()
```

Reply:
OK, converted to uppercase.